engine/src/main/java/org/hibernate/validator/internal/metadata/location/TypeArgumentConstraintLocation.java

```diff
@@ -113,8 +115,12 @@ public boolean equals(Object o) {
 
 	@Override
 	public int hashCode() {
+		return hashCode;
+	}
+
+	private static int buildHashCode(ConstraintLocation delegate, TypeVariable<?> typeParameter) {
 		int result = delegate != null ? delegate.hashCode() : 0;
```

Comment:
Oh, I thought you were checking with me that typeArgument cannot be null :). Yeah agreed, will fix.